### ISSUE TITLE
Problems with logging and LVish debug flag

### ISSUE THREAD

**Issue by osa1** [iu-parfunc/lvars#114]
Some of the tests depend on logging with `-fdebug`, but others break when `-fdebug` is enabled.

* `par-transformers` wants it.
* `lvish` breaks when it's enabled.
* `par-mergesort` takes forever when it's enabled.

Not sure about how to solve this yet.

**Comment by rrnewton** [comment]
Yeah, it seems like we would need the (nonexistent) cabal feature where we depend on a particular flag setting of the `lvish` package from the test-suite section.

Short of that I think we can only enforce this in our scripts, not in the cabal setup.

**Comment by osa1** [comment]
Scripts and Cabal are already causing enough problems.. I was wondering if we could fix this by not relying on logging in tests(unless the test is about logging, of course). We can easily use an IORef with `internalLiftIO` to collect logs. How does that sound?

**Comment by rrnewton** [comment]
Oh, that's fine too... so to kind of manually do logging in the tests.

On Tue, Jul 7, 2015 at 12:04 PM, Ömer Sinan Ağacan <notifications@github.com

**Comment by rrnewton** [comment]
Or, maybe we should follow a convention where we use different package name
for the debug version e.g. "lvish-debug".

Cabal doesn't directly support that either of course, so I guess it would
have to be a separate cabal file.-

**Comment by osa1** [comment]
I made some progress on this, tests are now passing when -fdebug is not used(Travis is green), but we're having some problems when it's used.
